index.js

```diff
@@ -111,13 +104,25 @@ module.exports = {
         // limit to only files in the sass directory.
         tree = funnel(tree, { include: [path.join(sassDir, '/**/*')] });
 
+        // TODO: this is legacy compat
         let projectConfig = addon.project.config(host.env);
         if (addon.parent && addon.parent.engineConfig) {
           projectConfig = addon.parent.engineConfig(host.env, projectConfig);
         }
 
-        // setup eyeglass for this project's configuration
-        const config = projectConfig.eyeglass ? cloneDeep(projectConfig.eyeglass) : {};
+        let config;
+        if (projectConfig.eyeglass) {
+          // TODO: WTF engines
+          let from = addon.parent.root + '/config/environment';
+          let to = addon.parent.root + '/ember-cli-build';
+          addon.ui.writeDeprecateLine(`'eyeglass' configuration within config/environment is no longer supported\n  please move this configuration:\n\tfrom: '${from}' (or however configured) \n\tto:   '${to}' (or however configured)\n`);
+          config = cloneDeep(projectConfig.eyeglass);
+        } else {
+          // setup eyeglass for this project's configuration
+          const hostConfig = cloneDeep(host.options.eyeglass || {});
+          const addonConfig = isNestedAddon ? cloneDeep(addon.parent.options.eyeglass || {}) : {};
+          config = defaultsDeep(addonConfig, hostConfig);
```

Comment:
This seems unrelated to migrating from `config/environment.js` to me (though the feature itself seems nice)...

Reply:
should i drop?